### PR TITLE
[AutoDiff] Move dumpActivityInfo to DifferentiableActivityInfo class

### PR DESCRIPTION
**Pull request by bgogul**
Minor refactoring as part of https://bugs.swift.org/browse/TF-993

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: include/swift/SILOptimizer/Analysis/DifferentiableActivityAnalysis.h [hunk 1]
```diff
@@ -206,6 +206,16 @@ class DifferentiableActivityInfo {
   /// Returns the activity of the given value for the given `SILAutoDiffIndices`
   /// (parameter indices and result index).
   Activity getActivity(SILValue value, const SILAutoDiffIndices &indices) const;
+
+  /// Prints activity information for the `indices` of the given `value`.
+  void dumpActivityInfo(SILValue value,
```
Comment (dan-zheng) [inline include/swift/SILOptimizer/Analysis/DifferentiableActivityAnalysis.h]: How about renaming these to just `dump`?

**Review by dan-zheng — APPROVED** [review]

File: include/swift/SILOptimizer/Analysis/DifferentiableActivityAnalysis.h (hunk 1 above)
Comment (bgogul) [reply include/swift/SILOptimizer/Analysis/DifferentiableActivityAnalysis.h]: Ah that is better. Done.

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Analysis/DifferentiableActivityAnalysis.cpp [hunk 2]
```diff
@@ -421,3 +421,32 @@ Activity DifferentiableActivityInfo::getActivity(
...
+  switch (activity.toRaw()) {
+  case 0: s << "NONE"; break;
+  case (unsigned)ActivityFlags::Varied: s << "VARIED"; break;
+  case (unsigned)ActivityFlags::Useful: s << "USEFUL"; break;
+  case (unsigned)ActivityFlags::Active: s << "ACTIVE"; break;
+  }
+  s << "] " << value;
+}
+
+void DifferentiableActivityInfo::dump(SILFunction &fn,
```
Comment (rxwei) [inline lib/SILOptimizer/Analysis/DifferentiableActivityAnalysis.cpp]: The function argument is unnecessary. `getFunction()` gives you the function.

Comment (bgogul) [reply lib/SILOptimizer/Analysis/DifferentiableActivityAnalysis.cpp]: Good catch. Done!

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by bgogul** [comment]
Tests pass. I will go ahead and merge this in.

**Pull request merged**